crates/aptos/src/common/types.rs

```diff
@@ -2152,6 +2174,75 @@ impl TransactionOptions {
         Ok(summary)
     }
 
+    pub async fn simulate_using_session(
+        &self,
+        session_path: &Path,
+        payload: TransactionPayload,
+    ) -> CliTypedResult<TransactionSummary> {
+        let mut sess = Session::load(session_path)?;
+
+        let state_store = sess.state_store();
+
+        // Fetch the chain states required for the simulation
+        // TODO(Gas): get the following from the chain
```

Comment:
How difficult is it to resolve this TODO?

Reply:
My bad, this is an outdated TODO -- we can already remove it